internal/deviations/deviations.go

```diff
@@ -65,27 +65,91 @@ package deviations
 
 import (
 	"flag"
+	"fmt"
+	"regexp"
 
 	log "github.com/golang/glog"
 	"github.com/openconfig/featureprofiles/internal/metadata"
 	mpb "github.com/openconfig/featureprofiles/proto/metadata_go_proto"
 	"github.com/openconfig/ondatra"
 )
 
-// lookupDutDeviations returns the deviations for the specified dut.
-func lookupDUTDeviations(dut *ondatra.DUTDevice) *mpb.Metadata_Deviations {
+func matchRegex(regex string, deviceInfo string) bool {
+	match, err := regexp.MatchString(regex, deviceInfo)
+	if err != nil {
+		log.Errorf("Error with regex match %v", err)
+	}
+	return match
+}
+
+func isPlatformExceptionsValid(platformExceptions *mpb.Metadata_PlatformExceptions) error {
+	if platformExceptions.GetPlatform().Vendor.String() == "" {
+		return fmt.Errorf("vendor should be specified in textproto %v", platformExceptions)
+	}
+	if platformExceptions.GetPlatform().GetHardwareModelRegex() != "" && len(platformExceptions.GetPlatform().GetHardwareModel()) > 0 {
+		return fmt.Errorf("vendor should be specified in textproto %v", platformExceptions)
+	}
+	return nil
+}
+
+func lookupDeviations(dut *ondatra.DUTDevice) (*mpb.Metadata_PlatformExceptions, error) {
+	matchedPlatformException := &mpb.Metadata_PlatformExceptions{}
```

Comment:
I'd prefer if this were initially nil and you use non-nil in place of isPlatformExceptionsMatched

Reply:
done